cmd/ls.go

```diff
@@ -43,28 +56,41 @@ func ls(cmd *cobra.Command, args []string) (err error) {
 	path := ""
 	if len(args) > 0 {
 		if path, err = validatePath(args[0]); err != nil {
-			return
+			return err
 		}
 	}
 
 	arg := files.NewListFolderArg(path)
 
 	res, err := dbx.ListFolder(arg)
+	var entries []*files.Metadata
 	if err != nil {
-		return
-	}
-
-	entries := res.Entries
-
-	for res.HasMore {
-		arg := files.NewListFolderContinueArg(res.Cursor)
+		// Don't treat a "not_folder" error as fatal; recover by sending a
+		// get_metadata request for the same path and using that response instead.
+		if strings.Contains(err.Error(), "path/not_folder/") {
```

Comment:
So the changes I want to make will take longer than I anticipated and I'd rather not block this commit. Can you rebase and then we can merge. Thanks!